File: pset6/mario.py
#!/usr/bin/env python3

import cs50
import sys

def main():
    # validate the height
    while (True):
        print("Enter the height of the pyramid: ")
        height = cs50.get_int()
        if (height > 23 or height < 0):
            print ("Invalid input try again\n")
        else:
            break
    
    space = height - 1
    hashs = 1
    i=0
    # An outer loop to manage the every row of the pyramid
    while(i < height):
        # display spaces on the left side
        j=0
        while(j < space):
            sys.stdout.write(" ")
            j+=1 
        # display hashes on the left side
        k=0
        while (k < hashs):
            sys.stdout.write("#")
            k+=1
        # create gap
        sys.stdout.write("  ")
        # display hashes on the right side
        k=0
        while (k < hashs):
            sys.stdout.write("#")
            k+=1
        print("\n");
        i+=1
        space-=1
        hashs+=1
if __name__ == "__main__":
    main()

File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>
#define MAX_HEIGHT 23
/* A program to display a symmetric pyramids which look like
   Nintendos classic mario game.
*/
int main(void)
{
    int height=-1;
    printf("Enter the height of the pyramid: ");
    // validate the height
    while(true)
    {
        height = get_int();
        if(height > MAX_HEIGHT || height < 0)
        {
            printf("Invalid input");
        }
        else
        {
            break;
        }
    }

    int space = height - 1;
    int hash = 1;
    // An outer loop to manage the every row of the pyramid
    for(int i = 0;i < height;i++,space--,hash++)
    {
        // display spaces on the left side
        for(int j = 0;j < space;j++)
        {
            printf(" ");
        }
        // display hashes on the left side
        for(int k = 0;k < hash;k++)
        {
            printf("#");
        }
        // create gap
        printf("  ");
        // display hashes on the right side
        for(int k = 0;k < hash;k++)
        {
            printf("#");
        }
        printf("\n");
   }
}   